megamek/src/megamek/client/generator/TeamLoadoutGenerator.java

```diff
@@ -652,15 +652,62 @@ public static ReconfigurationParameters generateParameters(
 
     // region Imperative mutators
     private static void setACImperatives(Entity e, MunitionTree mt, ReconfigurationParameters rp) {
+        setAC5Imperatives(e, mt, rp);
+        setAC10Imperatives(e, mt, rp);
         setAC20Imperatives(e, mt, rp);
     }
 
+    private static int getACWeaponCount(Entity e, String size) {
+        return (int) e.getWeaponList().stream()
+                .filter(w -> w.getName().toLowerCase().contains("ac") && w.getName().contains(size)).count();
+    }
+
+    private static int getACAmmoCount(Entity e, String size) {
+        return (int) e.getAmmo().stream()
+                .filter(w -> w.getName().toLowerCase().contains("ac") && w.getName().contains(size)).count();
+    }
+
+    // Set low-ammo-count AC5 carriers to use Caseless if ammo is <= 1/2 ton per tube
+    private static boolean setAC5Imperatives(Entity e, MunitionTree mt, ReconfigurationParameters rp) {
```

Comment:
This function is almost identical to the AC/10 (and probably the AC/2 and AC/20 one). Let's re-work it into one function with an extra "AC Rating" parameter.

Reply:
Normally I would agree, but I intentionally differentiated these, as an example of how I would like additional imperatives to be added.
I want it to be very simple to add a new imperative for a specific weapon type and size, even for novice programmers, and I want the flow in this section to be very clear.
And making a single function that takes the AC rating and branches on that would likely produce as much code as we have now, without being any clearer.

So I must respectfully decline.

Reply:
Not sure how thes approach below produces more code than copy+pasting setACImperatives 4 times. Following function calls is a basic programming skill that it's better that novices learn. I would also argue that exposing novice programmers to "copy + paste identical method with slight modification" programming technique rather than "pull common logic out into separate function" is not what we want to be teaching novice coders.

Finally, if you decide to change the "setACImperatives" logic across the board, you have to remember to change it in 3-4 different methods, rather than changing it in just the one. 

setAC2Imperatives(entity, munitiontree, reconfigurationparameters) {
 setACImperatives(entity, munitiontree, reconfigurationparameters, 2)
}

setAC5Imperatives(entity, munitiontree, reconfigurationparameters) {
 setACImperatives(entity, munitiontree, reconfigurationparameters, 5)
}

setAC20Imperatives(entity, munitiontree, reconfigurationparameters) {
 setACImperatives(entity, munitiontree, reconfigurationparameters, 20)
}

setAC10Imperatives(entity, munitiontree, reconfigurationparameters) {
 setACImperatives(entity, munitiontree, reconfigurationparameters, 10)
}

setACImperatives(entity, munitiontree, reconfigurationparameters, barrelsize) {
getACWeaponCount(barrelsize);
aerocheck() or no acweaponcount;
if (acammocount <= acbarrelcount) use caseless
use any
}

Anyway, I'll let someone else tie-break this one.

Reply:
Speaking as one of the aforementioned novice programmers, I am tilted towards a preference for single methods over multiple similar methods. The biggest reason is from a maintainer perspective: it's a lot easier to adjust behavior reliably when you only have the one place to look. Having multiple methods increases the likelihood of one getting missed during an update pass.

Now, I will say that both of you are far more experienced at this than I, so I really don't feel like I'm qualified to make a decision, but you brought me into this PR so I figured I'd offer my thoughts.